Mandriva Linux boot logo.

Original author is unknown.
(Was probably Nicolas Planel & Warly in contrib/kernel-2.6.0-0.1mdk in
Mandriva 9.2 Fivestar).

Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>

---
 drivers/video/logo/Kconfig              |    5 
 drivers/video/logo/Makefile             |    1 
 drivers/video/logo/logo.c               |    4 
 drivers/video/logo/logo_mdk_clut224.ppm |19204 ++++++++++++++++++++++++++++++++
 4 files changed, 19214 insertions(+)

--- a/drivers/video/logo/Makefile
+++ b/drivers/video/logo/Makefile
@@ -15,6 +15,7 @@ obj-$(CONFIG_LOGO_SUPERH_MONO)		+= logo_
 obj-$(CONFIG_LOGO_SUPERH_VGA16)		+= logo_superh_vga16.o
 obj-$(CONFIG_LOGO_SUPERH_CLUT224)	+= logo_superh_clut224.o
 obj-$(CONFIG_LOGO_M32R_CLUT224)		+= logo_m32r_clut224.o
+obj-$(CONFIG_LOGO_MDK_CLUT224)		+= logo_mdk_clut224.o
 
 obj-$(CONFIG_SPU_BASE)			+= logo_spe_clut224.o
 
--- /dev/null
+++ b/drivers/video/logo/logo_mdk_clut224.ppm
@@ -0,0 +1,19204 @@
+P3
+# CREATOR: The GIMP's PNM Filter Version 1.0
+80 80
+255
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+8
+8
+8
+21
+21
+21
+34
+34
+34
+66
+66
+66
+92
+92
+92
+113
+113
+113
+134
+134
+134
+145
+145
+145
+161
+161
+161
+168
+168
+168
+174
+174
+174
+168
+168
+168
+161
+161
+161
+145
+145
+145
+134
+134
+134
+113
+113
+113
+92
+92
+92
+66
+66
+66
+34
+34
+34
+21
+21
+21
+8
+8
+8
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+8
+8
+8
+34
+34
+34
+77
+77
+77
+113
+113
+113
+145
+145
+145
+174
+174
+174
+174
+174
+174
+174
+174
+174
+174
+174
+174
+177
+177
+177
+181
+181
+181
+186
+186
+186
+188
+188
+188
+190
+190
+190
+190
+190
+190
+190
+190
+190
+188
+188
+188
+186
+186
+186
+181
+181
+181
+181
+181
+181
+177
+177
+177
+174
+174
+174
+145
+145
+145
+113
+113
+113
+77
+77
+77
+34
+34
+34
+8
+8
+8
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+34
+34
+34
+77
+77
+77
+130
+130
+130
+158
+158
+158
+174
+174
+174
+174
+174
+174
+181
+181
+181
+190
+190
+190
+201
+201
+201
+208
+208
+208
+215
+215
+215
+220
+220
+220
+222
+222
+222
+226
+226
+226
+229
+229
+229
+230
+230
+230
+230
+230
+230
+229
+229
+229
+226
+226
+226
+222
+222
+222
+220
+220
+220
+215
+215
+215
+212
+212
+212
+205
+205
+205
+196
+196
+196
+190
+190
+190
+186
+186
+186
+166
+166
+166
+130
+130
+130
+77
+77
+77
+34
+34
+34
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+44
+44
+44
+103
+103
+103
+158
+158
+158
+174
+174
+174
+174
+174
+174
+186
+186
+186
+199
+199
+199
+212
+212
+212
+222
+222
+222
+230
+230
+230
+236
+236
+236
+241
+241
+241
+244
+244
+244
+246
+246
+246
+249
+249
+249
+250
+250
+250
+251
+251
+251
+252
+252
+252
+252
+252
+252
+251
+251
+251
+250
+250
+250
+248
+248
+248
+246
+246
+246
+244
+244
+244
+241
+241
+241
+236
+236
+236
+230
+230
+230
+222
+222
+222
+212
+212
+212
+205
+205
+205
+196
+196
+196
+188
+188
+188
+166
+166
+166
+109
+109
+109
+44
+44
+44
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+44
+44
+44
+109
+109
+109
+168
+168
+168
+174
+174
+174
+181
+181
+181
+196
+196
+196
+212
+212
+212
+226
+226
+226
+236
+236
+236
+244
+244
+244
+246
+246
+246
+250
+250
+250
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+249
+249
+249
+246
+246
+246
+241
+241
+241
+236
+236
+236
+226
+226
+226
+217
+217
+217
+205
+205
+205
+196
+196
+196
+181
+181
+181
+117
+117
+117
+44
+44
+44
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+92
+92
+92
+161
+161
+161
+174
+174
+174
+181
+181
+181
+201
+201
+201
+220
+220
+220
+234
+234
+234
+244
+244
+244
+248
+248
+248
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+251
+251
+251
+246
+246
+246
+241
+241
+241
+232
+232
+232
+222
+222
+222
+212
+212
+212
+199
+199
+199
+177
+177
+177
+103
+103
+103
+21
+21
+21
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+8
+8
+8
+58
+58
+58
+145
+145
+145
+174
+174
+174
+181
+181
+181
+201
+201
+201
+222
+222
+222
+236
+236
+236
+246
+246
+246
+250
+250
+250
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+249
+249
+249
+244
+244
+244
+236
+236
+236
+226
+226
+226
+212
+212
+212
+201
+201
+201
+158
+158
+158
+66
+66
+66
+8
+8
+8
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+103
+103
+103
+166
+166
+166
+174
+174
+174
+196
+196
+196
+217
+217
+217
+234
+234
+234
+246
+246
+246
+250
+250
+250
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+249
+249
+249
+244
+244
+244
+236
+236
+236
+222
+222
+222
+208
+208
+208
+190
+190
+190
+117
+117
+117
+21
+21
+21
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+34
+34
+34
+134
+134
+134
+174
+174
+174
+181
+181
+181
+208
+208
+208
+230
+230
+230
+244
+244
+244
+249
+249
+249
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+249
+249
+249
+241
+241
+241
+232
+232
+232
+217
+217
+217
+208
+208
+208
+158
+158
+158
+44
+44
+44
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+53
+53
+53
+145
+145
+145
+174
+174
+174
+193
+193
+193
+217
+217
+217
+236
+236
+236
+246
+246
+246
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+246
+246
+246
+239
+239
+239
+226
+226
+226
+212
+212
+212
+174
+174
+174
+66
+66
+66
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+70
+70
+71
+158
+158
+158
+174
+174
+174
+201
+201
+201
+226
+226
+226
+241
+241
+241
+249
+249
+249
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+249
+249
+249
+244
+244
+244
+232
+232
+232
+217
+217
+217
+188
+188
+188
+83
+83
+83
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+77
+77
+77
+161
+161
+161
+177
+177
+177
+205
+205
+205
+230
+230
+230
+244
+244
+244
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+236
+183
+251
+242
+211
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+250
+250
+250
+244
+244
+244
+234
+234
+234
+222
+222
+222
+199
+199
+199
+92
+92
+92
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+70
+70
+71
+161
+161
+161
+177
+177
+177
+208
+208
+208
+232
+232
+232
+246
+246
+246
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+247
+226
+249
+223
+130
+245
+200
+37
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+246
+246
+246
+236
+236
+236
+226
+226
+226
+201
+201
+201
+83
+83
+83
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+53
+53
+53
+158
+158
+158
+177
+177
+177
+208
+208
+208
+234
+234
+234
+246
+246
+246
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+229
+154
+248
+216
+105
+228
+185
+27
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+246
+246
+246
+239
+239
+239
+229
+229
+229
+196
+196
+196
+70
+70
+71
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+34
+34
+34
+149
+149
+149
+174
+174
+174
+205
+205
+205
+232
+232
+232
+246
+246
+246
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+236
+183
+250
+226
+141
+246
+205
+57
+211
+168
+15
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+248
+248
+248
+239
+239
+239
+229
+229
+229
+186
+186
+186
+44
+44
+44
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+134
+134
+134
+174
+174
+174
+201
+201
+201
+230
+230
+230
+246
+246
+246
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+248
+235
+250
+227
+147
+249
+220
+118
+243
+194
+17
+203
+161
+15
+240
+231
+202
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+246
+246
+246
+239
+239
+239
+229
+229
+229
+174
+174
+174
+21
+21
+21
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+8
+8
+8
+103
+103
+103
+174
+174
+174
+196
+196
+196
+226
+226
+226
+244
+244
+244
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+227
+147
+249
+224
+134
+247
+211
+84
+233
+186
+16
+203
+161
+15
+227
+216
+178
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+246
+246
+246
+239
+239
+239
+230
+230
+230
+134
+134
+134
+8
+8
+8
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+58
+58
+58
+166
+166
+166
+190
+190
+190
+222
+222
+222
+244
+244
+244
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+239
+197
+250
+226
+141
+248
+218
+111
+246
+204
+51
+230
+184
+16
+205
+163
+15
+217
+201
+133
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+246
+246
+246
+239
+239
+239
+222
+222
+222
+77
+77
+77
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+145
+145
+145
+181
+181
+181
+212
+212
+212
+239
+239
+239
+250
+250
+250
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+241
+250
+227
+147
+249
+222
+126
+247
+211
+77
+245
+199
+32
+230
+184
+16
+209
+166
+15
+216
+194
+116
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+250
+250
+250
+246
+246
+246
+239
+239
+239
+190
+190
+190
+34
+34
+34
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+92
+92
+92
+174
+174
+174
+201
+201
+201
+232
+232
+232
+248
+248
+248
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+219
+224
+235
+219
+224
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+229
+154
+249
+224
+134
+248
+214
+93
+245
+202
+45
+244
+196
+21
+233
+186
+16
+211
+168
+15
+190
+160
+47
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+250
+250
+250
+244
+244
+244
+239
+239
+239
+117
+117
+117
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+44
+44
+44
+161
+161
+161
+188
+188
+188
+222
+222
+222
+244
+244
+244
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+209
+216
+232
+131
+154
+200
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+241
+202
+250
+226
+141
+248
+218
+111
+246
+207
+64
+245
+198
+28
+244
+195
+18
+233
+186
+16
+213
+170
+15
+198
+158
+16
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+249
+249
+249
+244
+244
+244
+220
+220
+220
+53
+53
+53
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+109
+109
+109
+177
+177
+177
+208
+208
+208
+236
+236
+236
+249
+249
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+194
+205
+227
+61
+97
+168
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+227
+147
+249
+222
+126
+247
+211
+77
+245
+200
+37
+244
+196
+21
+244
+195
+18
+233
+186
+16
+213
+170
+15
+193
+153
+14
+253
+250
+241
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+248
+248
+248
+244
+244
+244
+152
+152
+152
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+44
+44
+44
+168
+168
+168
+193
+193
+193
+226
+226
+226
+246
+246
+246
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+219
+224
+235
+71
+105
+173
+249
+249
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+232
+165
+250
+225
+137
+248
+214
+96
+245
+202
+45
+244
+197
+25
+244
+195
+17
+244
+195
+18
+236
+189
+17
+217
+174
+15
+198
+158
+16
+252
+251
+245
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+251
+251
+251
+248
+248
+248
+234
+234
+234
+66
+66
+66
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+1
+1
+2
+103
+103
+103
+181
+181
+181
+212
+212
+212
+239
+239
+239
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+88
+119
+181
+183
+195
+220
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+236
+183
+251
+233
+170
+250
+239
+197
+252
+247
+226
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+244
+220
+250
+227
+147
+249
+220
+118
+246
+207
+64
+245
+199
+32
+244
+195
+18
+244
+195
+17
+244
+195
+17
+240
+192
+17
+224
+179
+16
+205
+163
+15
+234
+219
+162
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+252
+252
+252
+248
+248
+248
+149
+149
+149
+1
+1
+2
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+34
+34
+34
+158
+158
+158
+193
+193
+193
+226
+226
+226
+246
+246
+246
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+114
+139
+192
+55
+88
+156
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+247
+211
+84
+248
+218
+111
+249
+223
+130
+249
+223
+130
+249
+224
+134
+249
+224
+134
+250
+225
+137
+250
+232
+165
+250
+239
+197
+252
+247
+226
+253
+250
+241
+253
+252
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+249
+250
+232
+165
+249
+223
+130
+247
+211
+84
+245
+201
+42
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+242
+193
+17
+226
+181
+16
+211
+168
+15
+217
+201
+133
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+226
+226
+226
+44
+44
+44
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+77
+77
+77
+177
+177
+177
+208
+208
+208
+236
+236
+236
+250
+250
+250
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+161
+178
+213
+58
+93
+165
+244
+245
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+242
+193
+17
+245
+202
+45
+247
+211
+84
+248
+216
+105
+248
+218
+111
+248
+218
+111
+248
+218
+112
+249
+220
+118
+249
+222
+126
+249
+223
+130
+249
+224
+134
+250
+227
+147
+250
+232
+165
+251
+236
+183
+252
+241
+202
+252
+247
+226
+252
+241
+202
+250
+232
+165
+250
+226
+141
+248
+214
+96
+246
+204
+51
+244
+197
+25
+244
+195
+18
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+233
+186
+16
+213
+170
+15
+215
+174
+30
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+251
+251
+109
+109
+109
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+8
+8
+8
+130
+130
+130
+190
+190
+190
+222
+222
+222
+244
+244
+244
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+238
+239
+242
+98
+126
+186
+95
+119
+169
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+252
+252
+221
+177
+16
+235
+188
+16
+245
+199
+32
+246
+205
+57
+246
+207
+64
+247
+209
+72
+247
+211
+84
+248
+214
+96
+248
+216
+105
+248
+218
+111
+248
+218
+112
+249
+220
+118
+249
+221
+121
+249
+223
+130
+250
+226
+141
+250
+227
+147
+249
+222
+126
+248
+214
+96
+246
+205
+57
+245
+198
+28
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+237
+189
+17
+221
+177
+16
+207
+165
+15
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+188
+188
+188
+21
+21
+21
+0
+0
+0
+1
+1
+2
+1
+1
+2
+34
+34
+34
+161
+161
+161
+201
+201
+201
+232
+232
+232
+249
+249
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+139
+160
+204
+57
+92
+162
+238
+239
+242
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+242
+211
+217
+174
+15
+226
+181
+16
+237
+189
+17
+244
+196
+21
+245
+198
+28
+245
+200
+37
+245
+202
+45
+246
+204
+51
+246
+205
+57
+246
+207
+64
+247
+209
+72
+247
+211
+77
+246
+213
+91
+248
+214
+93
+246
+213
+91
+247
+209
+72
+245
+202
+45
+245
+198
+28
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+240
+192
+17
+230
+184
+16
+221
+177
+16
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+232
+232
+232
+53
+53
+53
+0
+0
+0
+1
+1
+2
+1
+1
+2
+77
+77
+77
+181
+181
+181
+212
+212
+212
+239
+239
+239
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+199
+207
+224
+103
+130
+188
+60
+89
+148
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+237
+215
+137
+221
+177
+16
+230
+184
+16
+237
+189
+17
+243
+194
+17
+244
+195
+17
+244
+195
+18
+244
+195
+18
+244
+196
+21
+244
+196
+21
+244
+197
+25
+245
+199
+32
+245
+200
+37
+245
+201
+42
+245
+200
+37
+245
+198
+28
+244
+196
+21
+244
+195
+18
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+195
+17
+242
+193
+17
+240
+192
+17
+244
+195
+18
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+241
+250
+244
+220
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+241
+251
+233
+170
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+113
+113
+113
+0
+0
+0
+1
+1
+2
+8
+8
+8
+117
+117
+117
+190
+190
+190
+222
+222
+222
+244
+244
+244
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+139
+160
+204
+58
+93
+165
+153
+168
+198
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+229
+191
+55
+217
+174
+15
+230
+184
+16
+237
+189
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+196
+21
+245
+199
+32
+245
+202
+45
+246
+208
+70
+250
+229
+154
+252
+241
+202
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+247
+209
+72
+253
+250
+241
+253
+253
+253
+253
+253
+253
+253
+253
+253
+248
+216
+105
+248
+216
+101
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+168
+168
+168
+8
+8
+8
+1
+1
+2
+21
+21
+21
+149
+149
+149
+199
+199
+199
+230
+230
+230
+248
+248
+248
+253
+253
+253
+253
+253
+253
+253
+253
+253
+204
+213
+233
+123
+146
+197
+52
+83
+147
+209
+216
+232
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+213
+170
+15
+217
+174
+15
+230
+184
+16
+240
+192
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+196
+21
+245
+199
+32
+245
+202
+45
+246
+208
+70
+248
+214
+93
+248
+218
+111
+249
+220
+118
+249
+221
+121
+250
+232
+165
+252
+247
+226
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+249
+248
+216
+105
+253
+253
+253
+253
+253
+253
+252
+251
+245
+246
+205
+57
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+212
+212
+212
+21
+21
+21
+1
+1
+2
+34
+34
+34
+177
+177
+177
+208
+208
+208
+236
+236
+236
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+161
+178
+213
+93
+123
+183
+51
+81
+142
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+213
+170
+15
+221
+177
+16
+233
+186
+16
+240
+192
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+196
+21
+245
+198
+28
+245
+202
+45
+246
+207
+64
+247
+211
+84
+248
+214
+96
+248
+216
+101
+248
+216
+105
+248
+218
+112
+249
+220
+118
+250
+225
+137
+251
+236
+183
+252
+247
+226
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+232
+165
+249
+220
+118
+253
+253
+253
+244
+197
+25
+250
+244
+220
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+248
+248
+248
+53
+53
+53
+1
+1
+2
+70
+70
+71
+188
+188
+188
+215
+215
+215
+241
+241
+241
+252
+252
+252
+253
+253
+253
+253
+253
+253
+244
+245
+249
+131
+154
+200
+61
+97
+168
+82
+105
+151
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+246
+240
+221
+215
+172
+15
+224
+179
+16
+235
+188
+16
+242
+193
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+196
+21
+244
+196
+21
+245
+199
+32
+245
+201
+42
+246
+204
+51
+246
+205
+57
+247
+209
+72
+246
+213
+91
+248
+214
+96
+248
+216
+105
+248
+218
+111
+249
+220
+118
+249
+220
+118
+250
+232
+165
+251
+236
+183
+251
+248
+235
+253
+253
+253
+247
+211
+84
+251
+236
+183
+248
+218
+112
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+99
+99
+99
+1
+1
+2
+99
+99
+99
+193
+193
+193
+222
+222
+222
+244
+244
+244
+253
+253
+253
+253
+253
+253
+253
+253
+253
+204
+212
+229
+123
+146
+197
+55
+89
+157
+104
+125
+168
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+234
+219
+162
+217
+174
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+196
+21
+244
+197
+25
+245
+199
+32
+245
+201
+42
+246
+204
+51
+246
+207
+64
+247
+211
+77
+247
+211
+84
+247
+211
+84
+247
+211
+77
+246
+208
+70
+245
+202
+45
+243
+194
+17
+246
+234
+189
+252
+251
+245
+251
+242
+211
+247
+227
+158
+249
+224
+134
+250
+225
+137
+248
+218
+112
+247
+230
+170
+253
+253
+253
+253
+253
+253
+134
+134
+134
+1
+1
+2
+117
+117
+117
+196
+196
+196
+226
+226
+226
+246
+246
+246
+253
+253
+253
+253
+253
+253
+253
+253
+253
+161
+178
+213
+114
+139
+192
+53
+86
+151
+139
+153
+182
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+216
+194
+116
+221
+177
+16
+233
+186
+16
+240
+192
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+244
+195
+18
+244
+196
+21
+244
+196
+21
+243
+194
+17
+237
+189
+17
+226
+181
+16
+215
+172
+15
+209
+166
+15
+207
+165
+15
+209
+166
+15
+250
+244
+220
+248
+238
+198
+251
+242
+211
+248
+238
+200
+248
+238
+198
+247
+236
+197
+246
+234
+189
+251
+248
+235
+253
+253
+253
+253
+253
+253
+166
+166
+166
+1
+1
+2
+145
+145
+145
+201
+201
+201
+230
+230
+230
+249
+249
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+123
+146
+197
+98
+126
+186
+52
+83
+147
+166
+178
+202
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+229
+191
+55
+230
+184
+16
+240
+192
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+240
+192
+17
+233
+186
+16
+224
+179
+16
+207
+165
+15
+188
+149
+13
+173
+137
+12
+169
+134
+13
+169
+134
+13
+227
+216
+178
+253
+253
+253
+221
+188
+70
+248
+238
+198
+246
+208
+70
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+193
+193
+193
+1
+1
+2
+158
+158
+158
+205
+205
+205
+232
+232
+232
+250
+250
+250
+253
+253
+253
+253
+253
+253
+253
+253
+253
+123
+146
+197
+88
+119
+181
+51
+81
+142
+161
+173
+197
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+237
+189
+17
+244
+196
+21
+244
+196
+21
+244
+195
+18
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+240
+192
+17
+233
+186
+16
+224
+179
+16
+207
+165
+15
+188
+149
+13
+169
+134
+13
+169
+134
+13
+169
+134
+13
+227
+216
+178
+253
+253
+253
+253
+253
+253
+234
+219
+162
+229
+191
+55
+253
+253
+253
+247
+230
+170
+236
+189
+17
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+215
+215
+215
+1
+1
+2
+174
+174
+174
+208
+208
+208
+236
+236
+236
+251
+251
+251
+253
+253
+253
+253
+253
+253
+244
+245
+249
+123
+146
+197
+84
+115
+178
+51
+81
+142
+161
+173
+197
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+249
+221
+121
+245
+202
+45
+245
+200
+37
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+240
+192
+17
+233
+186
+16
+224
+179
+16
+207
+165
+15
+188
+149
+13
+169
+134
+13
+169
+134
+13
+169
+134
+13
+228
+215
+170
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+221
+188
+70
+253
+253
+253
+253
+253
+253
+253
+253
+253
+229
+191
+55
+237
+215
+137
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+232
+232
+232
+1
+1
+2
+181
+181
+181
+212
+212
+212
+239
+239
+239
+252
+252
+252
+253
+253
+253
+253
+253
+253
+209
+216
+232
+123
+146
+197
+78
+110
+176
+49
+78
+138
+161
+173
+197
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+229
+154
+247
+211
+77
+246
+204
+51
+244
+197
+25
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+242
+193
+17
+235
+188
+16
+224
+179
+16
+207
+165
+15
+188
+149
+13
+169
+134
+13
+169
+134
+13
+177
+141
+14
+227
+216
+178
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+228
+185
+27
+247
+230
+170
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+241
+215
+174
+30
+252
+251
+245
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+244
+244
+244
+1
+1
+2
+188
+188
+188
+212
+212
+212
+239
+239
+239
+253
+253
+253
+253
+253
+253
+253
+253
+253
+194
+205
+227
+123
+146
+197
+73
+106
+174
+51
+81
+142
+161
+173
+197
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+225
+137
+248
+214
+96
+246
+205
+57
+244
+197
+25
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+240
+192
+17
+230
+184
+16
+213
+170
+15
+193
+153
+14
+169
+134
+13
+169
+134
+13
+169
+134
+13
+231
+224
+196
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+244
+220
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+249
+253
+250
+241
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+249
+249
+249
+1
+1
+2
+186
+186
+186
+215
+215
+215
+239
+239
+239
+253
+253
+253
+253
+253
+253
+253
+253
+253
+161
+178
+213
+114
+139
+192
+71
+105
+173
+51
+81
+142
+166
+178
+202
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+242
+211
+249
+222
+126
+248
+214
+96
+246
+204
+51
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+242
+193
+17
+230
+184
+16
+209
+166
+15
+177
+141
+14
+169
+134
+13
+169
+134
+13
+228
+215
+170
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+246
+246
+246
+1
+1
+2
+181
+181
+181
+217
+217
+217
+239
+239
+239
+253
+253
+253
+253
+253
+253
+253
+253
+253
+161
+178
+213
+103
+130
+188
+67
+102
+171
+51
+81
+142
+104
+125
+168
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+227
+147
+249
+222
+126
+247
+211
+77
+245
+200
+37
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+237
+189
+17
+217
+174
+15
+188
+149
+13
+169
+134
+13
+228
+212
+158
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+232
+232
+232
+1
+1
+2
+166
+166
+166
+215
+215
+215
+239
+239
+239
+252
+252
+252
+253
+253
+253
+253
+253
+253
+144
+164
+208
+98
+126
+186
+71
+105
+173
+52
+83
+147
+104
+125
+168
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+241
+202
+249
+224
+134
+248
+216
+105
+246
+205
+57
+244
+197
+25
+244
+195
+18
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+233
+186
+16
+207
+165
+15
+173
+137
+12
+228
+212
+158
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+215
+215
+215
+1
+1
+2
+149
+149
+149
+215
+215
+215
+236
+236
+236
+251
+251
+251
+253
+253
+253
+253
+253
+253
+161
+178
+213
+93
+123
+183
+73
+106
+174
+52
+83
+147
+82
+105
+151
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+225
+137
+249
+221
+121
+247
+211
+77
+245
+200
+37
+244
+196
+21
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+230
+184
+16
+203
+161
+15
+177
+141
+14
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+193
+193
+193
+1
+1
+2
+130
+130
+130
+212
+212
+212
+236
+236
+236
+250
+250
+250
+253
+253
+253
+253
+253
+253
+204
+212
+229
+88
+119
+181
+78
+110
+176
+53
+86
+151
+49
+78
+138
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+252
+241
+202
+249
+223
+130
+248
+216
+101
+246
+204
+51
+244
+197
+25
+244
+195
+18
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+230
+184
+16
+205
+163
+15
+190
+160
+47
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+166
+166
+166
+1
+1
+2
+109
+109
+109
+212
+212
+212
+232
+232
+232
+249
+249
+249
+253
+253
+253
+253
+253
+253
+209
+216
+232
+88
+119
+181
+84
+115
+178
+56
+90
+160
+49
+78
+138
+219
+224
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+225
+137
+249
+221
+121
+247
+211
+77
+245
+200
+37
+244
+195
+18
+244
+195
+17
+244
+195
+17
+244
+195
+17
+243
+194
+17
+240
+192
+17
+237
+189
+17
+233
+186
+16
+233
+186
+16
+237
+189
+17
+240
+192
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+233
+186
+16
+207
+165
+15
+190
+154
+27
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+134
+134
+134
+1
+1
+2
+83
+83
+83
+212
+212
+212
+232
+232
+232
+248
+248
+248
+253
+253
+253
+253
+253
+253
+253
+253
+253
+88
+119
+181
+88
+119
+181
+57
+92
+162
+51
+81
+142
+166
+178
+202
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+242
+211
+249
+224
+134
+248
+216
+101
+246
+204
+51
+244
+197
+25
+244
+195
+17
+244
+195
+17
+242
+193
+17
+237
+189
+17
+230
+184
+16
+221
+177
+16
+211
+168
+15
+207
+165
+15
+207
+165
+15
+217
+174
+15
+226
+181
+16
+237
+189
+17
+243
+194
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+233
+186
+16
+209
+166
+15
+188
+149
+13
+253
+252
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+99
+99
+99
+1
+1
+2
+44
+44
+44
+205
+205
+205
+229
+229
+229
+246
+246
+246
+253
+253
+253
+253
+253
+253
+253
+253
+253
+88
+119
+181
+88
+119
+181
+61
+97
+168
+53
+86
+151
+104
+125
+168
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+250
+226
+141
+249
+221
+121
+247
+211
+77
+245
+199
+32
+243
+194
+17
+240
+192
+17
+233
+186
+16
+226
+181
+16
+213
+170
+15
+196
+156
+14
+188
+149
+13
+173
+137
+12
+173
+137
+12
+188
+149
+13
+204
+162
+8
+213
+170
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+18
+233
+186
+16
+209
+166
+15
+188
+149
+13
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+249
+249
+249
+53
+53
+53
+1
+1
+2
+21
+21
+21
+177
+177
+177
+226
+226
+226
+244
+244
+244
+253
+253
+253
+253
+253
+253
+253
+253
+253
+103
+130
+188
+78
+110
+176
+67
+102
+171
+55
+89
+157
+52
+83
+147
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+236
+183
+249
+224
+134
+248
+216
+101
+245
+202
+45
+240
+192
+17
+230
+184
+16
+217
+174
+15
+205
+163
+15
+188
+149
+13
+173
+137
+12
+169
+134
+13
+169
+134
+13
+190
+160
+47
+227
+226
+149
+231
+224
+196
+218
+210
+180
+209
+173
+3
+211
+168
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+233
+186
+16
+211
+168
+15
+188
+149
+13
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+212
+212
+212
+21
+21
+21
+1
+1
+2
+8
+8
+8
+145
+145
+145
+222
+222
+222
+239
+239
+239
+251
+251
+251
+253
+253
+253
+253
+253
+253
+161
+178
+213
+67
+102
+171
+73
+106
+174
+58
+93
+165
+55
+88
+156
+153
+168
+198
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+249
+224
+134
+248
+218
+112
+245
+202
+45
+230
+184
+16
+211
+168
+15
+196
+156
+14
+177
+141
+14
+169
+134
+13
+169
+134
+13
+173
+137
+12
+216
+194
+116
+248
+246
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+219
+224
+235
+253
+253
+253
+204
+162
+8
+211
+168
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+244
+195
+17
+244
+195
+17
+244
+195
+17
+233
+186
+16
+213
+170
+15
+188
+149
+13
+251
+242
+211
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+168
+168
+168
+8
+8
+8
+1
+1
+2
+1
+1
+2
+92
+92
+92
+220
+220
+220
+236
+236
+236
+249
+249
+249
+253
+253
+253
+253
+253
+253
+204
+213
+233
+61
+97
+168
+67
+102
+171
+61
+97
+168
+57
+92
+162
+55
+89
+157
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+251
+236
+183
+248
+218
+111
+245
+201
+42
+213
+170
+15
+188
+149
+13
+169
+134
+13
+169
+134
+13
+169
+134
+13
+193
+166
+73
+227
+216
+178
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+236
+241
+251
+47
+84
+162
+3
+43
+195
+253
+253
+253
+253
+253
+253
+203
+161
+15
+213
+170
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+244
+195
+17
+244
+195
+17
+233
+186
+16
+213
+170
+15
+193
+153
+14
+238
+231
+202
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+113
+113
+113
+0
+0
+0
+1
+1
+2
+1
+1
+2
+44
+44
+44
+199
+199
+199
+232
+232
+232
+246
+246
+246
+253
+253
+253
+253
+253
+253
+253
+253
+253
+78
+110
+176
+59
+95
+168
+67
+102
+171
+59
+95
+168
+58
+93
+165
+84
+115
+178
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+246
+213
+91
+244
+196
+21
+198
+158
+16
+169
+134
+13
+169
+134
+13
+177
+141
+14
+216
+199
+141
+248
+246
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+123
+146
+197
+56
+90
+160
+82
+105
+151
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+205
+163
+15
+213
+170
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+244
+195
+17
+235
+188
+16
+213
+170
+15
+193
+153
+14
+246
+240
+221
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+232
+232
+232
+53
+53
+53
+0
+0
+0
+1
+1
+2
+1
+1
+2
+8
+8
+8
+161
+161
+161
+229
+229
+229
+244
+244
+244
+252
+252
+252
+253
+253
+253
+253
+253
+253
+161
+178
+213
+57
+92
+162
+59
+95
+168
+61
+97
+168
+61
+97
+168
+61
+97
+168
+139
+160
+204
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+249
+220
+118
+233
+186
+16
+193
+153
+14
+169
+134
+13
+193
+166
+73
+248
+246
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+183
+195
+220
+73
+106
+174
+53
+86
+151
+204
+212
+229
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+205
+163
+15
+213
+170
+15
+226
+181
+16
+237
+189
+17
+242
+193
+17
+235
+188
+16
+215
+172
+15
+196
+156
+14
+231
+224
+196
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+188
+188
+188
+21
+21
+21
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+99
+99
+99
+226
+226
+226
+239
+239
+239
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+55
+89
+157
+57
+92
+162
+58
+93
+165
+67
+102
+171
+73
+106
+174
+78
+110
+176
+131
+154
+200
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+247
+211
+77
+217
+174
+15
+227
+226
+149
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+199
+207
+224
+98
+126
+186
+56
+90
+160
+104
+125
+168
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+205
+163
+15
+213
+170
+15
+226
+181
+16
+237
+189
+17
+235
+188
+16
+217
+174
+15
+196
+156
+14
+228
+212
+158
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+253
+253
+253
+113
+113
+113
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+34
+34
+34
+201
+201
+201
+236
+236
+236
+248
+248
+248
+253
+253
+253
+253
+253
+253
+253
+253
+253
+183
+195
+220
+55
+88
+156
+56
+90
+160
+58
+93
+165
+71
+105
+173
+84
+115
+178
+93
+123
+183
+103
+130
+188
+204
+213
+233
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+219
+224
+235
+170
+186
+219
+103
+130
+188
+57
+92
+162
+52
+83
+147
+219
+224
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+205
+163
+15
+213
+170
+15
+226
+181
+16
+233
+186
+16
+217
+174
+15
+196
+156
+14
+228
+212
+158
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+230
+230
+230
+44
+44
+44
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+1
+1
+2
+134
+134
+134
+232
+232
+232
+246
+246
+246
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+95
+119
+169
+53
+86
+151
+55
+88
+156
+58
+93
+165
+71
+105
+173
+88
+119
+181
+103
+130
+188
+114
+139
+192
+123
+146
+197
+194
+205
+227
+244
+245
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+249
+249
+249
+204
+213
+233
+170
+186
+219
+131
+154
+200
+93
+123
+183
+56
+90
+160
+49
+78
+138
+199
+207
+224
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+205
+163
+15
+213
+170
+15
+221
+177
+16
+215
+172
+15
+196
+156
+14
+228
+215
+170
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+152
+152
+152
+1
+1
+2
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+58
+58
+58
+222
+222
+222
+241
+241
+241
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+60
+89
+148
+51
+81
+142
+52
+83
+147
+55
+88
+156
+58
+93
+165
+78
+110
+176
+98
+126
+186
+114
+139
+192
+131
+154
+200
+144
+164
+208
+161
+178
+213
+161
+178
+213
+161
+178
+213
+144
+164
+208
+139
+160
+204
+131
+154
+200
+103
+130
+188
+59
+95
+168
+52
+83
+147
+46
+73
+129
+139
+153
+182
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+205
+163
+15
+211
+168
+15
+209
+166
+15
+196
+156
+14
+216
+194
+116
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+246
+246
+246
+66
+66
+66
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+145
+145
+145
+236
+236
+236
+248
+248
+248
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+51
+81
+142
+46
+73
+129
+49
+78
+138
+51
+81
+142
+53
+86
+151
+56
+90
+160
+59
+95
+168
+78
+110
+176
+93
+123
+183
+98
+126
+186
+88
+119
+181
+73
+106
+174
+58
+93
+165
+55
+88
+156
+51
+81
+142
+46
+73
+129
+46
+73
+129
+161
+173
+197
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+203
+161
+15
+198
+158
+16
+188
+149
+13
+216
+194
+116
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+158
+158
+158
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+53
+53
+53
+217
+217
+217
+244
+244
+244
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+161
+173
+197
+42
+67
+119
+42
+67
+119
+46
+73
+129
+46
+73
+129
+46
+73
+129
+46
+73
+129
+46
+73
+129
+46
+73
+129
+46
+73
+129
+42
+67
+119
+42
+67
+119
+42
+67
+119
+139
+153
+182
+219
+224
+235
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+188
+149
+13
+177
+141
+14
+216
+194
+116
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+236
+236
+236
+58
+58
+58
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+117
+117
+117
+239
+239
+239
+249
+249
+249
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+199
+207
+224
+82
+105
+151
+42
+67
+119
+42
+67
+119
+42
+67
+119
+42
+67
+119
+42
+67
+119
+82
+105
+151
+166
+178
+202
+252
+252
+252
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+177
+141
+14
+216
+194
+116
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+134
+134
+134
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+34
+34
+34
+193
+193
+193
+244
+244
+244
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+208
+208
+208
+34
+34
+34
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+83
+83
+83
+232
+232
+232
+250
+250
+250
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+244
+244
+244
+83
+83
+83
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+8
+8
+8
+145
+145
+145
+246
+246
+246
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+152
+152
+152
+8
+8
+8
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+190
+190
+190
+251
+251
+251
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+196
+196
+196
+34
+34
+34
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+53
+53
+53
+212
+212
+212
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+215
+215
+215
+53
+53
+53
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+83
+83
+83
+226
+226
+226
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+226
+226
+226
+83
+83
+83
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+99
+99
+99
+236
+236
+236
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+236
+236
+236
+103
+103
+103
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+109
+109
+109
+236
+236
+236
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+236
+236
+236
+109
+109
+109
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+99
+99
+99
+226
+226
+226
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+229
+229
+229
+103
+103
+103
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+77
+77
+77
+212
+212
+212
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+212
+212
+212
+83
+83
+83
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+58
+58
+58
+196
+196
+196
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+196
+196
+196
+58
+58
+58
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+152
+152
+152
+241
+241
+241
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+244
+244
+244
+152
+152
+152
+21
+21
+21
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+8
+8
+8
+83
+83
+83
+205
+205
+205
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+205
+205
+205
+83
+83
+83
+8
+8
+8
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+34
+34
+34
+134
+134
+134
+234
+234
+234
+254
+254
+254
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+234
+234
+234
+134
+134
+134
+34
+34
+34
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+58
+58
+58
+158
+158
+158
+244
+244
+244
+254
+254
+254
+254
+254
+254
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+244
+244
+244
+158
+158
+158
+58
+58
+58
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+66
+66
+66
+152
+152
+152
+229
+229
+229
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+229
+229
+229
+152
+152
+152
+66
+66
+66
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+44
+44
+44
+113
+113
+113
+188
+188
+188
+232
+232
+232
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+253
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+232
+232
+232
+188
+188
+188
+113
+113
+113
+44
+44
+44
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+21
+21
+21
+53
+53
+53
+113
+113
+113
+168
+168
+168
+212
+212
+212
+252
+252
+252
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+254
+252
+252
+252
+212
+212
+212
+168
+168
+168
+113
+113
+113
+53
+53
+53
+21
+21
+21
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+0
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+8
+8
+8
+21
+21
+21
+53
+53
+53
+99
+99
+99
+134
+134
+134
+168
+168
+168
+193
+193
+193
+215
+215
+215
+232
+232
+232
+246
+246
+246
+250
+250
+250
+246
+246
+246
+232
+232
+232
+215
+215
+215
+193
+193
+193
+168
+168
+168
+134
+134
+134
+99
+99
+99
+53
+53
+53
+21
+21
+21
+8
+8
+8
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
+1
+1
+2
--- a/drivers/video/logo/Kconfig
+++ b/drivers/video/logo/Kconfig
@@ -82,4 +82,9 @@ config LOGO_M32R_CLUT224
 	depends on M32R
 	default y
 
+config LOGO_MDK_CLUT224
+        bool "224-color Mandriva logo"
+        depends on LOGO
+        default y
+
 endif # LOGO
--- linux-2.6.31-rc3-git5-mnb1/drivers/video/logo/logo.c.video-mdk-logo.orig	2009-07-22 01:11:13.000000000 +0300
+++ linux-2.6.31-rc3-git5-mnb1/drivers/video/logo/logo.c	2009-07-23 00:19:07.000000000 +0300
@@ -63,6 +63,11 @@ const struct linux_logo * __init_refok f
 	}
 	
 	if (depth >= 8) {
+#ifdef CONFIG_LOGO_MDK_CLUT224
+		/* Mandriva logo */
+		logo = &logo_mdk_clut224;
+		return logo;
+#endif
 #ifdef CONFIG_LOGO_LINUX_CLUT224
 		/* Generic Linux logo */
 		logo = &logo_linux_clut224;
--- linux-2.6.30/include/linux/linux_logo.h.orig	2009-07-23 01:34:25.000000000 +0300
+++ linux-2.6.30/include/linux/linux_logo.h	2009-07-23 02:47:16.000000000 +0300
@@ -47,6 +47,7 @@ extern const struct linux_logo logo_supe
 extern const struct linux_logo logo_superh_clut224;
 extern const struct linux_logo logo_m32r_clut224;
 extern const struct linux_logo logo_spe_clut224;
+extern const struct linux_logo logo_mdk_clut224;
 
 extern const struct linux_logo *fb_find_logo(int depth);
 #ifdef CONFIG_FB_LOGO_EXTRA
